language/stdlib/modules/validator_config_2.move

```diff
@@ -0,0 +1,142 @@
+address 0x0:
+
+module ValidatorConfig2 {
+    use 0x0::Transaction;
+    use 0x0::LibraAccount;
+
+    struct Config {
+        consensus_pubkey: vector<u8>,
+        // This sequence_number equals to the number of key rotations done for this validator's account.
+        // Note that this number is different from the LibraAccount::T.sequence_number as
+        // here the key rotations for this account may be sent this account as well
+        // by delegated_address, if it is present. In both cases, the sequence_number will be
+        // incremented.
+        sequence_number: u64,
+    }
+
+    // A current or prospective validator should publish one of these under their address.
+    resource struct T {
+        config: Config,
+        managing_delegated: bool,
+        delegated_address: address,
+    }
```

Comment:
do we want events here?
maybe something out of band listens for events and pushes a reconfiguration if it is deemed necessary?

Reply:
I left a todo to add events. Let me know if you think it's better to make them part of this PR.